src/actions/download/index.js

```diff
@@ -0,0 +1,79 @@
+import { ListObjectsCommand, S3Client } from '@aws-sdk/client-s3';
+import { createDir, writeBinaryFile } from '@tauri-apps/api/fs';
+import { basename, join } from '@tauri-apps/api/path';
+import { PROJECTS_ARCHIVE_PREFIX } from '../../constants/paths';
+import {
+  addIllustration,
+  addProject,
+  getProject,
+  getStoreValue,
+} from '../../store';
+import { downloadS3Object } from '../../utils/download-from-s3';
+import { fetchProjectMeta } from './project-meta';
+
+export async function downloadProject(projectSlug) {
+  const bucket = import.meta.env.VITE_AWS_BACKUP_BUCKET_NAME;
+  const s3 = new S3Client({
+    region: 'us-east-1',
+    credentials: {
+      accessKeyId: import.meta.env.VITE_AWS_KEY_ID,
+      secretAccessKey: import.meta.env.VITE_AWS_SECRET_KEY,
+    },
+  });
+
+  const keyPath = await join(PROJECTS_ARCHIVE_PREFIX, projectSlug);
+  const projectListCommand = new ListObjectsCommand({
+    Bucket: bucket,
+    Prefix: keyPath,
+  });
+
+  // Grabs file names in project folder
+  const { Contents } = await s3.send(projectListCommand);
+  const files = await Promise.all(
+    Contents.map(async (d) => {
+      const name = await basename(d.Key, '.ai');
+      return {
+        illoName: name,
+        key: d.Key,
+      };
+    }),
+  );
+
+  // Gets text file for project name
+  // This should be a JSON file that stores
+  // the illustration names as well
+  const projectName = await fetchProjectMeta(s3, files);
+  await addProject(projectName, { isUploaded: true });
+
+  // Start of downloading illustrator files
+  const projectsFolder = await getStoreValue('projectsFolder');
+  const projectPath = await join(projectsFolder, projectSlug);
+
+  try {
+    const illos = await Promise.all(
+      files
+        .filter(({ illoName }) => illoName !== 'project-name.txt')
+        .map(async ({ key, illoName }) => {
+          const destination = await join(projectPath, illoName);
+          const illoPath = await join(destination, `${illoName}.ai`);
+          const arr = await downloadS3Object({
+            key,
+            bucket,
+          });
+          await createDir(destination, { recursive: true });
+          await writeBinaryFile(illoPath, arr);
+          return illoName;
+        }),
+    );
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const illustrationName of illos) {
+      // eslint-disable-next-line no-await-in-loop
+      await addIllustration({ projectSlug, illustrationName });
+    }
```

Comment:
Tried that, but as of right now I have to fully wait for the prior illustration to be added to add the next one, since that's how the store does the insertion. I considered changing the `addIllustration` function to allow for an array of illustration names to be passed — then do the looping on the that side. But since this is the only place in the app that will be adding multiple illustrations to the store at one time, it didn't feel worth the effort.

But if you don't want the linter rule exceptions, I'll refactor it. Seems like that rule only exists to support older browsers, which doesn't apply here. So I'm open either way.

Reply:
Hmm, how do you feel about this utility I made a while back. It's basically promise.all but it resolves sequentially. 

https://politico.slack.com/archives/G01QRGH7XM2/p1628015237003300

Reply:
I don't mind that lint rule so much, but in general we don't really use for loops anywhere else these days.